python/lsst/ctrl/mpexec/singleQuantumExecutor.py

```diff
@@ -450,6 +465,15 @@ def writeLogRecords(self, quantum, taskDef, butler):
             # Remove the handler so we stop accumulating log messages.
             logging.getLogger().removeHandler(self.log_handler)
 
+        if not store:
+            # do not store anything but remove file
+            if filename is not None:
```

Comment:
This duplication logic really makes me wish that python had a `if`/`else`/`finally` clause so we could do:
```
if store:
    # store it
else:
    pass
finally:
    # remove file
```

Reply:
although clearly I'm being silly:

```python
if store and other tests:
    # store it

if filename:
    # remove it
```
would work wouldn't it?